### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by maleadt**
Trying to address recent issues and comments. @musm and @denizyuret, please review/test/fix.
I'm happy to have Windows support, but I can't test this code so it is almost guaranteed to contain errors.
It'll be up to you guys to get/keep it working; I don't get paid to do this 🙂 

Also, if there's compatibility issues (#17) please add tests so that we can avoid bugs in the code quering the version dictionaries (which needs an overhaul, hence this request).

### CODE REVIEW

**Comment by denizyuret** [comment]
I don't think they can pay us enough to support this s*&!t :)
I don't have a windows-gpu setup either, I get one from Amazon AWS for $1/hr to test.  Here is my AMI if you'd like to try:
- aws region = ohio (us-east-2)
- ami name = Knet171213win (ami-da4a62bf)
- instance type = p2.xlarge
- rdp login = Administrator / Knet-Windows
Testing your latest fixes now...

File: src/discovery.jl [hunk 1]
```diff
@@ -295,83 +290,81 @@ function find_host_compiler(toolkit_version=nothing)
...
-                msvc_path = joinpath(dirname(dirname(dirname(path))), "VC", "bin", arch, "cl.exe")
-                push!(msvc_paths, msvc_path)
-            end
-        catch ex
-            error("Compatible Visual Studio installation cannot be found; Visual Studio 2017, 2015, 2013, 2012, or 2010 is required.")
+        # discover Visual Studio installations
+        msvc_paths = String[]
+        arch = Sys.WORD_SIZE == 64 ? "amd64" : "x86"
+        ## locate VS2017
+        vswhere = joinpath(get(ENV, "ProgramFiles(x86)", ""), "Microsoft Visual Studio", "Installer", "vswhere.exe")
```
Comment (musm) [inline src/discovery.jl]: On 32 bit windows this should  just be `joinpath(get(ENV, "ProgramFiles", "")` although I can't test.

**Comment by maleadt** [comment]
Apparently the latest MSVC is incompatible with CUDA again (version 19.12.25830).

**Comment by maleadt** [comment]
OK, going to merge this. Tested on Windows for a change. Let me know if it works with your packages.
Does change the public API slightly, so will need an upper bound + fixes when we tag it. I plan to do that early next week, unless other issues arise.

**Pull request merged**